posts/introduction-to-grpc.md

```diff
@@ -0,0 +1,197 @@
+client is using, all it needs is the serialized message (which it knows how to deserialize).
+
+Now that we’ve discussed what gRPC is, let’s flesh out an example. We will create a simple example 
+application (available on GitHub - https://github.com/hegek87/derivative-solver) to compute the derivative 
+of simple polynomials. If you don’t remember how a derivative is calculated, we will quickly show the 
+general rule our server will use.
+
+## Polynomials and Derivatives
+
+Remember that a polynomial is a function of the form 
+> P(x) = c_nx^n + c_{n-1}x^{n-1} + …. + c_1x + c_0, 
+> 
+> where c_1, c_2, …, c_n are simple constants.
+
+To give a concrete example, let’s use the polynomial `x^4 + 2x^2 - x - 1`. Next, let’s get a better 
+understanding of how to find the derivative of a polynomial. If you never learned this, it’s a pretty 
+robotic procedure that will hopefully be straightforward to understand. To calculate a derivative, we
+simply multiply the exponent by the coefficient, and subtract one from the original exponent. Given 
+a simple polynomial, `4x^2`, we find the derivative by multiplying the exponent and coefficient, 
+`4*2 = 8`, then we subtract one from the exponent, `2-1=1`, giving us the derivative, `8x^1 = 8x`. 
+This procedure works for simple polynomials, and can be expanded for polynomials of more than one term.
+As another example, consider `x^3 + 3x^2 + 3`. Remember that x^0 = 1, so `3 = 3x^0`, and we can 
+calculate the derivative as `3*0 = 0`. The full derivative of the equations leads to `3x^2 + 6x + 0 = 3x^2 + 6x`.
+
+
+## Creating A Message Format
+Now that we understand what a derivative is, let’s discuss how we represent a polynomial for our protobuf.
+It seems like we have two simple ways to represent the polynomial, a naive way (which we handle in our rest
+call) to do this is by sending it as a raw string. While this works, it could be cumbersome, as we will
+need to find a way to split the string, parse the constants and exponents into numeric values and map back 
+to a string after calculating the derivative. Perhaps a more elegant solution is to send the constant as an
+array, and the exponents as an array, this way we can get the constant values and exponent values without 
+any parsing on our side. Something like
+
+```java
+Message Polynomial {
+	repeated coefficients = 1;
+	repeated exponents = 2;
+}
+```
+
+Now that we have our message structure defined, we can compile to see the generated code. As expected, we
+will see POJOs generated with setters, getters, and builders. Now that we've configured our message structure,
+we can define our service method that will be called in order to actually make requests:
+```java
+service DerivativeSolver {
+  rpc findDerivative(Polynomial) returns (Polynomial) {}
+}
+```
+
+Compiling using protoc once again, we will now have everything we need to set up our simple application. 
```

Comment:
Further down you present some code, I would start here adding the command used to compile, so even a reader that don't know much about gRPC (like me) can compile your code.